File: estimator/src/Services/customerService.ts
import { RateModel } from '../Models/RateModel';
import { CustomerModel } from '../Models/CustomerModel';
import { RateTypeModel } from '../Models/RateTypeModel';
import { HttpClient, HttpClientConfiguration, json } from 'aurelia-fetch-client';

export class CustomerService {
    public getCustomers(): Promise<Array<CustomerModel>> {
        let httpClient = new HttpClient();
        httpClient.configure(config => {
            config.useStandardConfiguration()
                .withBaseUrl('api/');
        });

        return httpClient.fetch('customers', { method: 'get' })
            .then(response => response.json())
            .then(responseJson => {
                let customers = new Array<CustomerModel>();
                responseJson as Array<any>;
                responseJson.forEach(customerJson => {
                    let customer = new CustomerModel(customerJson.name, []);
                    customerJson.rates.forEach(rate => {
                        customer.rates.push(new RateModel(rate.dailyRate, rate.code, null))
                    });
                    customers.push(customer);
                });
                return customers;
            });
    }
}

File: estimator/Services/CustomerService.cs
using System.Collections.Generic;
using estimator.web.Models;

namespace estimator.web.Services
{

    public class CustomerService
    {

        public IEnumerable<Customer> GetAllCustomers()
        {
            var ratesA = new List<Rate>() { new Rate { Code = "DT", DailyRate = 200 }, new Rate { Code = "TC", DailyRate = 300 }, new Rate { Code = "LTC", DailyRate = 400 }, new Rate { Code = "PM", DailyRate = 350 } };
            var ratesB = new List<Rate>() { new Rate { Code = "DT", DailyRate = 2000 }, new Rate { Code = "TC", DailyRate = 3000 }, new Rate { Code = "LTC", DailyRate = 4000 }, new Rate { Code = "PM", DailyRate = 3500 } };
            
            return new List<Customer>(){new Customer{ Name="Rob", Rates = ratesA}, new Customer{ Name="Bob", Rates= ratesB}};
        }
    }
}